packages/synthetic-chain/src/lib/core-eval-support.ts

```diff
@@ -150,13 +153,13 @@ export const ensureISTForInstall = async (
 export const readBundles = async (dir: string) => {
   const files = await fsp.readdir(dir);
   const names = files.filter(f => f.endsWith('.js')).map(f => f.slice(0, -3));
-  const buildAssets = {} as Record<string, BundleInfo>;
+  const bundleInfos: BundleInfo[] = [];
```

Comment:
Why change this type? This is the only breaking change I see. Is it necessary?

I factored it that way to enforce that there's on bundle per name, an assumption I think the user will have. Admittedly, given that goal the loop should have enforced it.

Reply:
> Why change this type? This is the only breaking change I see. Is it necessary?

Correct, the return value of `readBundles` and the param of `passCoreEvalProposal` are the only breaking changes. I didn't see them used anywhere so figured it was safe to break. It's necessary to break `passCoreEvalProposal` because it needs the directory information to come from somewhere. Before it was assuming a `submission` subdir

> I factored it that way to enforce that there's on bundle per name, an assumption I think the user will have. Admittedly, given that goal the loop should have enforced it.

As you noticed, the change from `bundleAssets` -> `bundleInfos` doesn't change the logic at all. While it does technically allow multiple bundle infos to share the same name, I don't think it's a huge problem if that's the case?

Reply:
> While it does technically allow multiple bundle infos to share the same name, I don't think it's a huge problem if that's the case?

Not huge, but worse. As I think about this more, there's no good reason to export this functionality from the package so I'm not concerned with the breaking change. I'll probably follow up with another breaking change (affecting nobody) to drop these exports. So refactor at will.

I might also add then a validation that names aren't reused, unless you want to add that now.